### ISSUE TITLE
Add option for disabling `identifier_name` and `type_name` for raw identifiers

### ISSUE THREAD

**Issue by cbarnes-rv** [realm/SwiftLint#6767]
### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Feature or Enhancement Proposal

With the addition of raw identifiers, it has become common to use more verbose names for test cases and test suites. For these test cases and test suites, names often will cause linting to fail:

```swift
@Suite
struct `Array.append(_:)` Tests` {
  @Test
  func `test append(_:) in empty array` {
    // Test code here
  }
}
```

It is possible to loosen the rules for these cases, but that may not always be ideal. In the previous example, to get the linting to pass (without many ignore comments), would require allowing `_` as an allowed character, which may not be ideal for other identifiers.

Another option is to use different linting profiles for tests, though this can be tedious when a product is divided into many smaller packages, each with a separate tests directory.

 I propose adding a new option for the `identifier_name` and `type_name` rules named `ignore_raw_identifiers`, which will disable the rule(s) when using a raw identifier (an identifier beginning and ending with a backtick).

**Comment by respan** [comment]
In Xcode 27, Apple added agent skills that can be exported and used elsewhere. In Test Modernizer skill, they explicitly recommend using raw identifiers when naming tests, here’s [an example](https://github.com/wisdom-in-a-nutshell/agents/blob/main/skills-source/external/test-modernizer/SKILL.md#test-methods).

If such an option is added, it will be very convenient.

**Comment by SimplyDanny** [comment]
Turns out, all the `*_name` rules already support an `excluded` configuration option which accepts a list of regular expressions. All identifiers that match, will be excluded from analysis. You can use it to exclude raw identifiers already now for `type_name` and `generic_type_name`. `identifier_name` requires some adaptions but will allow this in the same way eventually with #6775.

**Comment by SimplyDanny** [comment]
> Turns out, all the `*_name` rules already support an `excluded` configuration option which accepts a list of regular expressions. All identifiers that match, will be excluded from analysis. You can use it to exclude raw identifiers already now for `type_name` and `generic_type_name`. `identifier_name` requires some adaptions but will allow this in the same way eventually with [#6775](https://github.com/realm/SwiftLint/pull/6775).

In fact, we won't need an additional option.
